app/views/articles/_full_comment_area.html.erb

```diff
@@ -59,6 +59,7 @@
 
         <div class="comments" id="comment-trees-container">
           <% if @article.comments_count > 0 %>
+            <% @current_comment_index = 0 %>
```

Comment:
I was curious to see if Rails provide some kind of iteration helper in-place and it does. 

In this collection partial, the child class can use `comment_node_iteration.index` but the caveat is that it starts at `0` and we would need to pass the var down from `_comment_tree.html.erb`. The trade-off is arguable but it would be cleaner and avoid any potential off-by-one-error.

Reply:
@maestromac I explored this logic but didn't quite understand how it worked so I moved on, and still don't quite understand with your comment.

So I'd say if you want to take a cut at refactoring this post-merge you should have at it.